spec/services/hyrax/multiple_membership_checker_spec.rb

```diff
@@ -40,64 +40,65 @@
-        expect(Collection).to receive(:where).once.and_return(collections)
+        expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: [collection_type.gid]).once.and_return(collections)
         expect(subject).to eq 'Error: You have specified more than one of the same single-membership collection types: Greedy (Foo and Bar)'
       end
+
+      context 'with multiple single membership collection types' do
+        let!(:collection_type_2) { create(:collection_type, title: 'Doc', allow_multiple_membership: false) }
+        let(:collection_types) { [collection_type.gid, collection_type_2.gid] }
+
+        it 'returns an error' do
+          expect(item).not_to receive(:member_of_collection_ids)
+          expect(checker).to receive(:single_membership_collections).with(collection_ids).once.and_call_original
+          expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: collection_types).once.and_call_original
+          expect(subject).to eq 'Error: You have specified more than one of the same single-membership collection types: Greedy (Foo and Bar)'
+        end
+      end
     end
 
     context 'when multiple single-membership collection instances are in the list, including current members' do
-      let(:collection1) { create(:collection, title: ['Foo']) }
-      let(:collection2) { create(:collection, title: ['Bar']) }
+      let(:collection1) { create(:collection, title: ['Foo'], collection_type: collection_type) }
+      let(:collection2) { create(:collection, title: ['Bar'], collection_type: collection_type) }
       let(:collections) { [collection1] }
       let(:collection_ids) { collections.map(&:id) }
       let(:included) { true }
 
       before do
-        allow(collection1).to receive(:collection_type_gid).and_return(collection_type.gid)
-        allow(collection2).to receive(:collection_type_gid).and_return(collection_type.gid)
         allow(item).to receive(:member_of_collection_ids).once.and_return([collection2.id])
       end
 
-      it 'returns nil' do
+      it 'returns an error' do
         expect(item).to receive(:member_of_collection_ids)
-        expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: collection_type.gid).once.and_return(collections)
-        expect(Collection).to receive(:where).with(id: [collection2.id], collection_type_gid_ssim: collection_type.gid).once.and_return([collection2])
+        expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: [collection_type.gid]).once.and_return(collections)
+        expect(Collection).to receive(:where).with(id: [collection2.id], collection_type_gid_ssim: [collection_type.gid]).once.and_return([collection2])
         expect(subject).to eq 'Error: You have specified more than one of the same single-membership collection types: Greedy (Foo and Bar)'
       end
     end
```

Comment:
I would like to see a test case:
- col_type_1 is single_membership
- col_type_2 is single_membership
- col_1 is of type col_type_1
- col_2 is of type col_type_2
- checker allows both to be added since they are not of the same collection type

Reply:
:+1: I can add this.